lib/rubygems/installer.rb

```diff
@@ -538,14 +538,15 @@ def generate_plugins # :nodoc:
   def generate_bin_script(filename, bindir)
     bin_script_path = File.join bindir, formatted_program_filename(filename)
 
-    Gem.open_file_with_flock("#{bin_script_path}.lock") do
+    Gem.open_file_with_flock("#{bin_script_path}.lock") do |lock|
       require "fileutils"
       FileUtils.rm_f bin_script_path # prior install may have been --no-wrappers
 
       File.open(bin_script_path, "wb", 0o755) do |file|
         file.write app_script_text(filename)
         file.chmod(options[:prog_mode] || 0o755)
       end
+      File.unlink(lock.path)
```

Comment:
Should this go in an ensure in `open_file_with_flock` ?

Reply:
Good catch!

Reply:
Not only the lock file, the executable script seems left, after an error occurred.
This should be removed as well, and windows .bat files?